### PR TITLE
Add Portlibrary Filestream API

### PR DESCRIPTION
**Pull request by youngar**

### CODE REVIEW

File: port/common/omrfilestream.c [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+ * Determines the proper portable error code to return given a native error code.
+ *
+ * @param[in] errorCode The error code reported by the OS
+ *
+ * @return the (negative) portable error code
+ */
+static int32_t
+findError (int32_t errorCode)
+{
+	switch (errorCode)
```
Comment (alin00) [inline port/common/omrfilestream.c]: Please follow coding standard for formatting switch statements: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-switch-statements

File: fvtest/porttest/omrfilestreamTest.cpp [hunk 2]
```diff
@@ -0,0 +1,2692 @@
...
+	if (0 != rc) {
+		/* read from the file when it should be in the buffer.  Not sure if the c standard guarantees
+		 * it will respect your buffering wishes */
+		outputErrorMessage(PORTTEST_ERROR_ARGS, "file_read_all() returned %d expected %d\n", rc, 0);
+		goto test5CloseFile;
+	}
+
+	rc = omrfilestream_sync(writeFile);
+	if (0 > rc) {
+		outputErrorMessage(PORTTEST_ERR
```
Comment (alin00) [inline fvtest/porttest/omrfilestreamTest.cpp]: should be uint16_t here and line 1914 below

File: fvtest/porttest/omrfilestreamTest.cpp [hunk 3]
```diff
@@ -0,0 +1,2692 @@
...
+ *
+ * @return TEST_PASS on success, TEST_FAIL on error
+ */
+TEST(PortFileStreamTest, omrfilestream_test_buffering)
+{
+	const char *testName = "omrfilestream_test_buffering";
+	const char *filename = "omrfilestream_test_buffering.tst";
+	const char output[]  = "testing\noutput";
+	char input[]         = "xxxxxxxxxxxxxx";
+	I_32 characterCount = sizeof(output);
```
Comment (alin00) [inline fvtest/porttest/omrfilestreamTest.cpp]: use int32_t instead of I_32

File: port/common/omrfilestream.c [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+		portLibrary->error_set_last_error(portLibrary, EINVAL, findError(EINVAL));
+		Trc_PRT_filestream_fdopen_invalidArgs(fd, flags);
+		Trc_PRT_filestream_fdopen_Exit(NULL);
+		return NULL;
+	}
+
+	/* Get the native file descriptor */
+	fileDescriptor = portLibrary->file_convert_omrfile_fd_to_native_fd(portLibrary, fd);
+#if defined(J9ZOS390)
+	switch (fileDescriptor) {
```
Comment (charliegracie) [inline port/common/omrfilestream.c]: What is the rationale for the empty swtich?

**Comment by youngar** [comment]
@alin00 I un-indented the switch statement and replaced all the usages of I_* types in omrfilestreamTest.cpp (the other files were fine).

@charliegracie Its a mistake, the logic for that switch was moved into file_convert_omrfile_fd_to_native_fd and file_convet_native_fd_to_omr_fd.  I have deleted the empty switch.

File: port/unix/omrfile.c [hunk 5]
```diff
@@ -1237,11 +1237,31 @@ intptr_t
 omrfile_convert_native_fd_to_omrfile_fd(struct OMRPortLibrary *portLibrary, intptr_t nativeFD)
 {
 
-	intptr_t omrfileFD = nativeFD;
+#if (FD_BIAS != 0)
+	/* Do NOT add the FD_BIAS to standard streams */
+	switch (nativeFD) {
+		case OMRPORT_TTY_IN:
```
Comment (charliegracie) [inline port/unix/omrfile.c]: This switch is formatted incorrectly.  Can you fix it please?

File: port/win32/omrfile.c [hunk 6]
```diff
@@ -58,6 +58,26 @@ toHandle(OMRPortLibrary *portLibrary, intptr_t fd)
...
+/* Convert windows file handle to a file descriptor.
+ * OMRPORT_TTY_IN, OMRPORT_TTY_OUT and OMRPORT_TTY_ERR are handled as a special case because they don't
+ * match UNIX standard stream file descriptors.
+ * @param[in] fd returned by port library functions
+ * @return windows file handle.
+ */
+static intptr_t
+fromHandle(OMRPortLibrary *portLibrary, HANDLE handle)
+{
+	if (handle == PPG_tty_consoleInputHd)
```
Comment (charliegracie) [inline port/win32/omrfile.c]: Can you add the "{}" brackets for the if statements please?

File: port/win32/omrfilestream.c [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+ * @param[in] errorCode The error code reported in errno
+ *
+ * @return the (negative) portable error code
+ */
+static int32_t
+findErrorFromErrno (int32_t errorCode)
+{
+	switch (errorCode)
+	{
+		case EACCES:
```
Comment (charliegracie) [inline port/win32/omrfilestream.c]: Can you fix the formatting please? Including any fall through cases.

**Comment by youngar** [comment]
@charliegracie Updated the commit with your suggestions

**Pull request merged**